src/Concordium/Client/Runner.hs

```diff
@@ -2500,6 +2365,683 @@ generateBakerKeys bkBakerId = do
+processBakerConfigureCmd :: Maybe FilePath -> Verbose -> Backend -> TransactionOpts (Maybe Types.Energy)
+  -> Bool -- ^Whether this was called via `baker configure`
+  -> Maybe Types.Amount -- ^New stake/capital.
+  -> Maybe Bool -- ^Select whether to restake earnings.
+  -> Maybe Types.OpenStatus -- ^Open for delegation status.
+  -> Maybe String -- ^URL to metadata about baker.
+  -> Maybe Types.AmountFraction -- ^Transaction fee commission.
+  -> Maybe Types.AmountFraction -- ^Baking reward commission.
+  -> Maybe Types.AmountFraction -- ^Finalization commission.
+  -> Maybe FilePath -- ^File to read baker keys from.
+  -> Maybe FilePath -- ^File to write baker keys to.
+  -> IO ()
+processBakerConfigureCmd baseCfgDir verbose backend txOpts isBakerConfigure cbCapital cbRestakeEarnings cbOpenForDelegation metadataURL cbTransactionFeeCommission cbBakingRewardCommission cbFinalizationRewardCommission inputKeysFile outputKeysFile = do
+  let intOpts = toInteractionOpts txOpts
+  (bakerKeys, txCfg, pl) <- transactionForBakerConfigure (ioConfirm intOpts)
+  withClient backend $ do
+    when isBakerConfigure $ warnAboutMissingAddBakerParameters txCfg
+    mapM_ (warnAboutBadCapital txCfg) cbCapital
+    result <- sendAndTailTransaction txCfg pl intOpts
+    events <- eventsFromTransactionResult result
+    mapM_ (tryPrintKeyUpdateEventToOutputFile bakerKeys) events
+  where
+    warnAboutMissingAddBakerParameters txCfg = do
+      let allPresent = case (cbOpenForDelegation, metadataURL, cbTransactionFeeCommission, cbBakingRewardCommission, cbFinalizationRewardCommission, cbRestakeEarnings, cbCapital, inputKeysFile) of
+                (Just _, Just _, Just _, Just _, Just _, Just _, Just _, Just _) -> True
+                _ -> False
+      when (not allPresent) $ do
+        let senderAddr = naAddr . esdAddress . tcEncryptedSigningData $ txCfg
+        Types.AccountInfo{..} <- getAccountInfoOrDie senderAddr
+        case aiStakingInfo of
+          Types.AccountStakingBaker{} -> return ()
+          _ -> do
+            logWarn $ [init ("To add a baker, more options are necessary. The following are missing "
+                  ++ (if isNothing cbCapital then "\n--stake," else "")
+                  ++ (if isNothing cbOpenForDelegation then "\n--open-delegation-for," else "")
+                  ++ (if isNothing inputKeysFile then "\n--keys-in," else "")
+                  ++ (if isNothing metadataURL then "\n--baker-url," else "")
+                  ++ (if isNothing cbTransactionFeeCommission then "\n--delegation-transaction-fee-commission," else "")
+                  ++ (if isNothing cbBakingRewardCommission then "\n--delegation-baking-commission," else "")
+                  ++ (if isNothing cbFinalizationRewardCommission then "\n--delegation-finalization-commission," else ""))
+                  ++ (if isNothing cbRestakeEarnings then ". \nExactly one of the options --restake and --no-restake must be present" else "")]
```

Comment:
Argument handling should be dealt with in `Commands.hs` when possible. My comment in that file has a suggested fix.

Reply:
This is not possible, since we are looking up whether the account is a baker or not.